### ISSUE TITLE
Enumeration of user ids in API endpoint BadgeUserEmailDetail

### ISSUE THREAD

**Issue by sveeke** [edubadges/audit#28]
It is possible to enumerate the user ids using this simple oracle in `/user/emails/<id>`:

Excerpt from class `BadgeUserEmailDetail()`:
```
        email_address = self.get_email(pk=id)
        if email_address is None:
            return Response(status=status.HTTP_404_NOT_FOUND)
        if email_address.user_id != self.request.user.id:
            return Response(status=status.HTTP_403_FORBIDDEN)
```

Impact: an attacker can find out which user ids are valid, this could be used in later steps of a more complex attack.

Recommendation: Return the same error code in both cases.

**Comment by ottonomy** [comment]
These ids are email ids, not user ids, so an attacker could only use this to find out which email ids are valid, or potentially triangulate on a rough guess at number of email addresses registered. 👍 will implement the simple fix. Fixed in eba3016c4b85dcbdbcd9cc77aa37a7d2e4171d7b

**Comment by sveeke** [comment]
Fixed
